src/Core/SettingsChangesHistory.h

```diff
@@ -92,6 +92,7 @@ static std::map<ClickHouseVersion, SettingsChangesHistory::SettingsChanges> sett
               {"cross_join_min_bytes_to_compress", 0, 1_GiB, "A new setting."},
               {"prefer_external_sort_block_bytes", 0, DEFAULT_BLOCK_SIZE * 256, "Prefer maximum block bytes for external sort, reduce the memory usage during merging."},
               {"input_format_force_null_for_omitted_fields", false, false, "Disable type-defaults for omitted fields when needed"},
+              {"output_format_pretty_preserve_border_for_multiline_string", 1, 1, "Applies better rendering for multiline strings."},
```

Comment:
Shouldn't previous value be `0` not `1`? If I understand correctly, to achieve previous behaviour we should disable this setting, so previpusly ClickHouse behaived like this setting is disabled?
P.S. SettingsChangesHistory is used to set settings to the state they were in some previous version.

Reply:
I thought that if we created a new setting, it doesn't matter what value we put as a previous value for this setting, because this setting simply did not exist before. Should I fix this in follow-up pr?

Reply:
> I thought that if we created a new setting, it doesn't matter what value we put as a previous value for this setting, because this setting simply did not exist before

Yes, it's a new setting, but this setting enables new behaviour by default. And when we set `compatibility` setting to some old version, we want the server to behave like this new setting wasn't added (so, in our case we should disable this new setting):
https://github.com/ClickHouse/ClickHouse/blob/6c0450f8ffd30d9d069e8dea1c78f05e6042135f/src/Core/SettingsChangesHistory.h#L83-L84

> Should I fix this in follow-up pr?

Yes, please.

Reply:
Got it, thanks, will fix it🙂

Reply:
described in #63783